router/src/server.rs

```diff
@@ -1643,6 +1645,69 @@ async fn embed(
     Ok(Json(response))
 }
 
+/// Embed inputs
+#[utoipa::path(
+    post,
+    tag = "OpenAI Compatible",
+    path = "/v1/embeddings",
+    request_body = CompatEmbedRequest,
+    responses(
+    (status = 200, description = "Embeddings ids", body = CompatEmbedResponse),
+    (status = 500, description = "Incomplete embedding", body = ErrorResponse),
+    )
+)]
+#[instrument(skip_all)]
+#[axum::debug_handler]
+async fn compat_embed(
+    infer: Extension<Infer>,
+    Json(req): Json<CompatEmbedRequest>,
+) -> Result<Json<CompatEmbedResponse>, (StatusCode, Json<ErrorResponse>)> {
+    metrics::increment_counter!("lorax_request_count");
+    tracing::debug!("Input: {}", req.input);
+    if let StringOrVec::Vec(inputs) = req.input {
```

Comment:
OOC what's the difference between string or vec as inputs? Is it the vec just a list of strings and why we have a .map call?

Reply:
yeah but we need to handle the case that the input is a string or a list of strings. Rust is a bit weird like that. So you need a special type for that. https://platform.openai.com/docs/api-reference/embeddings/create#embeddings-create-input

Reply:
Very interesting, good to know!